### PR TITLE
[yaml] backtick generated aliases on sql mappings

### PR DESCRIPTION
**Pull request by Polber**
There are cases where `MapToFields` does not work for sql expressions where a field in the incoming schema is a reserved keyword.

For example, with an input schema of 
`{foo: int, timestamp: str}`

The following would fail:
```
- type: MapToFields
  config:
    language: sql
    append: true
    fields:
      newField: foo
```
as the generated query is 
`SELECT (timestamp) AS timestamp, (foo) AS foo, (foo) AS newField FROM PCOLLECTION.`

where `(timestamp) AS timestamp` was generated due to `append: true` being set.

---

This PR provides a fix by auto-escaping aliases that are generated by the framework such as the example above, or when naming a field, i.e.
```
- type: MapToFields
  name: Transform
  config:
    language: sql
    fields:
      timestamp: foo
```

Note: The expression itself will not be auto-backticked, so the following is still expected to fail:
```
- type: MapToFields
  config:
    language: sql
    fields:
      foo: timestamp
```
a user would be expected to write valid SQL, i.e.
```
- type: MapToFields
  config:
    language: sql
    fields:
      foo: "`timestamp`"
```

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by Polber** [comment]
R: @damccorm

File: sdks/python/apache_beam/yaml/yaml_mapping.py
Comment (liferoad) [inline sdks/python/apache_beam/yaml/yaml_mapping.py:518]: looks like these issues should be easily caught with some tests. So can we add more tests?

File: sdks/python/apache_beam/yaml/tests/maptofields.yaml
Comment (damccorm) [inline sdks/python/apache_beam/yaml/tests/maptofields.yaml:56]: Is there a reason we can't just auto-backtick this as well? Is it basically so that we can support valid SQL expressions?

We should probably add to the docs what is/isn't acceptable here

File: sdks/python/apache_beam/yaml/yaml_mapping.py
Comment (Polber) [reply sdks/python/apache_beam/yaml/yaml_mapping.py]: Added some tests

File: sdks/python/apache_beam/yaml/tests/maptofields.yaml
Comment (Polber) [reply sdks/python/apache_beam/yaml/tests/maptofields.yaml]: Right, if someone has a schema that contains a reserved keyword, we don't want to override the function by auto-backticking in case they actually wanted to perform that function on a different field (or the same).

As far as documenting, I'm in agreement the more docs, the better, but I'm not really sure what would need to be explicitly called out here. The query is a **valid** SQL statement (i.e. reserved keywords are escaped when using as column names), and this PR doesn't really change that. It just makes it to where the YAML tags work as expected

Comment (Polber) [reply sdks/python/apache_beam/yaml/tests/maptofields.yaml]: On second thought, I added some more docs as they were lacking a lot of valuable information.

Comment (liferoad) [reply sdks/python/apache_beam/yaml/tests/maptofields.yaml]: what is the error message when this happens? Do we tell users know how to fix it?

File: website/www/site/content/en/documentation/sdks/yaml-udf.md
Comment (damccorm) [inline website/www/site/content/en/documentation/sdks/yaml-udf.md:189]:
```suggestion
If one wanted to select a field that collides with a [reserved SQL keyword](https://calcite.apache.org/docs/reference.html#keywords), the field(s) must be surrounded in backticks. For example, say the incoming PCollection has a field "timestamp", one would have to write:
```

**Review by damccorm — COMMENTED** [review]
LGTM pending one tiny comment and checks passing

File: website/www/site/content/en/documentation/sdks/yaml-udf.md
Comment (damccorm) [reply website/www/site/content/en/documentation/sdks/yaml-udf.md:189]: FYI, this is still pending (and blocking merging)

Comment (damccorm) [reply website/www/site/content/en/documentation/sdks/yaml-udf.md:189]: Note there are also now conflicts which need resolved

File: sdks/python/apache_beam/yaml/tests/maptofields.yaml
Comment (Polber) [reply sdks/python/apache_beam/yaml/tests/maptofields.yaml]: Just the generic SQL syntax error thrown by SQLTransform which I think makes sense given that it needs to be valid SQL (which is also documented now)

File: website/www/site/content/en/documentation/sdks/yaml-udf.md
Comment (Polber) [reply website/www/site/content/en/documentation/sdks/yaml-udf.md:189]: Done.

File: website/www/site/content/en/documentation/sdks/yaml-udf.md
Comment (damccorm) [inline website/www/site/content/en/documentation/sdks/yaml-udf.md:189]:
```suggestion
If one wanted to select a field that collides with a [reserved SQL keyword](https://calcite.apache.org/docs/reference.html#keywords), the field(s) must be surrounded in backticks. For example, say the incoming PCollection has a field "timestamp", one would have to write:
```

I'm just going to apply this so that it gets in before the cut

**Review by damccorm — APPROVED** [review]
LGTM, test failure is unrelated

**Pull request merged**
